CMakeLists.txt

```
@@ -426,32 +442,49 @@ if(MINGW)
 endif()
 
 set(lego1_objects)
-set(lego1_link_libraries dxguid d3drm_guid)
+set(lego1_link_libraries)
+if(WIN32)
+  list(APPEND lego1_link_libraries dxguid d3drm_guid)
+endif()
 foreach(lego1_library IN LISTS lego1_targets)
   target_compile_definitions(${lego1_library}-objects PRIVATE LEGO1_DLL)
   list(APPEND lego1_objects $<TARGET_OBJECTS:${lego1_library}-objects>)
   list(APPEND lego1_link_libraries ${lego1_library}-interface)
 endforeach()
-add_library(lego1 SHARED
-  LEGO1/main.cpp
-  ${lego1_objects}
-)
+if (CMAKE_BUILD_TYPE STREQUAL "Release")
+  add_library(lego1 SHARED
+    LEGO1/main.cpp
+    ${lego1_objects}
+  )
+else()
+  add_library(lego1 STATIC
+    LEGO1/main.cpp
+    ${lego1_objects}
+  )
+endif()
```

Id keep this `SHARED` for now, and turn everything into a single executable at a later point.
Since the goal of this repo is no longer to be accurate, the cmake script can be simplified greatly.
```suggestion
add_library(lego1 SHARED
  LEGO1/main.cpp
  ${lego1_objects}
)
```

---

This might also be the reason for the link failure on ci.

---

It's needed for ASAN